### DISCUSSION TITLE
Enforce removal of trailing slashes and auto-redirect within pages:extend hook?

### DISCUSSION THREAD

**Discussion by nathanchase** [nuxt/nuxt#17128] [Questions]
With Nuxt 3, is it possible to do a redirect directly inside a `pages:extend` hook? 

I'm trying to use [Vue's router options to strictly enforce no trailing slash](https://router.vuejs.org/api/interfaces/routeroptions.html#strict) and using [ufo's withoutTrailingSlash](https://github.com/unjs/ufo/#withouttrailingslash) and rewrite all Nuxt pages to eliminate all trailing slashes. 

Ideally this could just be a simple `trailingSlash: false` parameter in `nuxt.config` [like we have in Nuxt 2](https://nuxtjs.org/docs/configuration-glossary/configuration-router/#trailingslash).

![image](https://user-images.githubusercontent.com/5963574/183321643-0b161e2c-e5ee-41a5-bb12-31d1f7fc147e.png)

**Comment by nathanchase** [comment]
This seems to work, but I'm not sure if it's the best way to go about it...

```js
// ./middleware/notrailingslash.global.ts

import { withoutTrailingSlash } from 'ufo';

export default defineNuxtRouteMiddleware(to => {
  if (/\/+$/.test(to.path)) {
    return navigateTo(withoutTrailingSlash(to.path, true), { redirectCode: 301 });
  }
});

```

**Comment by benoitdemaegdt** [comment]
+1 for this question.
I asked more or less the same question in [@nuxt/content about this trailing slash](https://github.com/nuxt/content/discussions/1410).
I'm still not sure about the best way to deal with it 😞
